### ISSUE TITLE
Incorrect changelog created for self referencing domain

### ISSUE THREAD

**Issue by smakela** [grails-plugins/grails-database-migration#156]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Run example application

### Expected Behaviour

Link table should have two columns instead of one

### Actual Behaviour

Link table is created with only one column and insert fails.

org.springframework.jdbc.BadSqlGrammarException: Hibernate operation: could not prepare statement; bad SQL grammar [insert into foo_sub_foos (foo_id, sub_foos__id) values (?, ?)]; nested exception is org.h2.jdbc.JdbcSQLException: Column "FOO_ID" not found; SQL statement:
insert into foo_sub_foos (foo_id, sub_foos__id) values (?, ?) [42122-197]

Example works correctly if you change Hibernate to create the schema (create-drop for datasource and setting database migration settings to false). So bug lies maybe in the liquibase-hibernate5 library.

### Environment Information

- **Operating System**: OSX 10.14
- **Grails Version:** 3.3.9
- **Plugin Version:** 3.0.4
- **Database:** H2 or MySQL
- **JDK Version:** 1.8.0_181-b13

### Example Application

- https://github.com/smakela/migration-bug

**Comment by niravassar** [comment]
I tried reproducing this issue and was successful, both with Grails 4, grails 3.3.9 and also with the newest database-migration plugin `3.1.0.RC1`. Also tried to force the latest versions of liquibase-libraries. The error occurs in every case.

I briefly looked into trying to debug the issue and i think the problem starts in  `DiffCommand` of `liquibase-core:3.6.1`. I did not get to the point of exactly finding out what is wrong. I believe this bug should be filed in the liquibase project.

As a workaround, it should be easy as adding some DSL statements for the missing column into the `changelog.groovy`. 

Thank you
